### PR TITLE
KYP-761: Track product option ID if SKU is not present

### PR DESCRIPTION
**Pull request by nedick007**
Fallback logic to track product option ID if same product SKU is not available.

### CODE REVIEW

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready for review

File: Observer/AddToCart.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ private function addToCart($product, $quantity)
         if ($product->getId() != $childProduct->getId()) {
             $product = $this->productRepository->getById($product->getId());
             // for legacy reasons - we have been passing the SKU as ID for the child products
-            $data['option_id'] = $childProduct->getSku();
+            $data['option_id'] = $childProduct->getSku() ? $childProduct->getSku() : $childProduct->getId();
```
Comment (evgeni-spasov-metrilo) [inline Observer/AddToCart.php]: Lets make it consistent with the previous [pull request](https://github.com/Metrilo/metrilo-magento2-integration/pull/73/files#diff-3239ea2def5b5dfe803295a77f337128R18-R20).
```suggestion
            $sku = $childProduct->getSku();
            $data['option_id'] = $sku() ? $sku() : $childProduct->getId();
```

Comment (nedick007) [reply Observer/AddToCart.php]: Refactored.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo bump

File: Observer/AddToCart.php [hunk 2]
```diff
@@ -70,7 +70,8 @@ private function addToCart($product, $quantity)
         if ($product->getId() != $childProduct->getId()) {
             $product = $this->productRepository->getById($product->getId());
             // for legacy reasons - we have been passing the SKU as ID for the child products
-            $data['option_id'] = $childProduct->getSku();
+            $optionSku = $childProduct->getSku();
+            $data['option_id'] = $optionSku ? $optionSku : $childProduct->getId();
             $data['option_sku'] = $childProduct->getSku();
```
Comment (evgeni-spasov-metrilo) [inline Observer/AddToCart.php]: Use `$optionSku`. You have it already.

File: Observer/RemoveFromCart.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ public function execute(Observer $observer)
             $item = $observer->getEvent()->getQuoteItem();
             $product = $item->getProduct();
 
-            $this->helper->addSessionEvent('track', 'remove_from_cart', ['id' => $product->getId()]);
+            $this->helper->addSessionEvent('track', 'remove_from_cart', ['id' => $product->getSku() ? $product->getSku() : $product->getId()]);
```
Comment (evgeni-spasov-metrilo) [inline Observer/RemoveFromCart.php]: `$product->getSku()` called twice.

File: Observer/AddToCart.php (hunk 2 above)
Comment (nedick007) [reply Observer/AddToCart.php]: Fixed.

File: Observer/RemoveFromCart.php (hunk 3 above)
Comment (nedick007) [reply Observer/RemoveFromCart.php]: Fixed.

**Comment by nedick007** [comment]
@evgeni-spasov-metrilo ready

**Pull request merged**